Project/src/MakeCall/VolumeVisualizer.js

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import { LocalAudioStream } from '@azure/communication-calling';
+
+export default class VolumeVisualizer extends React.Component {
+    constructor(props) {
+        super(props);
+        this.call = props.call;
+        this.deviceManager = props.deviceManager;
+        this.state = {
+            localVolumeLevel: 0,
+            remoteVolumeLevel: 0,
+            localVolumeLevelSubscription: undefined,
+            remoteVolumeLevelSubscription: undefined,
+            localVolumeIndicator: undefined,
+            remoteVolumeIndicator: undefined,
+        };
+    }
+
+    async componentWillMount() {
+        if (this.call) {
+            let localVolumeStateSetter = undefined;
+            let handleSelectedMicrophoneVolumeSubscription = async () => {        
+                let localVolumeIndicator = await (new LocalAudioStream(this.deviceManager.selectedMicrophone).getVolume());
+                localVolumeStateSetter = ()=>{
+                    this.setState({ localVolumeLevel: localVolumeIndicator.level });
+                }
+                localVolumeIndicator.on('levelChanged', localVolumeStateSetter);
+                this.setState({ localVolumeLevelSubscription: localVolumeStateSetter });
```

Comment:
nit: we can move localVolumeLevelSubscription outside of state object to the class level, because it is not related to react state.